File: app/src/main/java/ml/docilealligator/infinityforreddit/UploadedImage.java
package ml.docilealligator.infinityforreddit;

import android.os.Parcel;
import android.os.Parcelable;

public class UploadedImage implements Parcelable {
    public String imageName;
    public String imageUrlOrKey;
    private String caption = "";

    public UploadedImage(String imageName, String imageUrlOrKey) {
        this.imageName = imageName;
        this.imageUrlOrKey = imageUrlOrKey;
    }

    protected UploadedImage(Parcel in) {
        imageName = in.readString();
        imageUrlOrKey = in.readString();
        caption = in.readString();
    }

    public String getCaption() {
        return caption == null ? "" : caption;
    }

    public void setCaption(String caption) {
        this.caption = caption == null ? "" : caption;
    }

    public static final Creator<UploadedImage> CREATOR = new Creator<UploadedImage>() {
        @Override
        public UploadedImage createFromParcel(Parcel in) {
            return new UploadedImage(in);
        }

        @Override
        public UploadedImage[] newArray(int size) {
            return new UploadedImage[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(imageName);
        parcel.writeString(imageUrlOrKey);
        parcel.writeString(caption);
    }
}

File: app/src/main/java/ml/docilealligator/infinityforreddit/UploadedImage.kt
package ml.docilealligator.infinityforreddit

import android.os.Parcel
import android.os.Parcelable

open class UploadedImage(
    var imageName: String?,
    var imageUrl: String?
    ) : Parcelable {
    protected constructor(parcel: Parcel) : this(
        parcel.readString(), parcel.readString()
    )

    override fun describeContents() = 0

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(imageName)
        parcel.writeString(imageUrl)
    }

    companion object CREATOR: Parcelable.Creator<UploadedImage?> {
        override fun createFromParcel(parcel: Parcel) =
            UploadedImage(parcel)
        override fun newArray(size: Int) =
            arrayOfNulls<UploadedImage?>(size)
    }
}
